### PR TITLE
Updating docs to support for APCu as a cache driver

### PR DESCRIPTION
**Pull request by reypm**

### CODE REVIEW

File: docs/en/reference/caching.rst [hunk 1]
```diff
@@ -63,6 +63,24 @@ by itself.
...
+
+Below is a simple example of how you could use the APCu cache driver
+by itself.
+
+.. code-block:: php
+
+    <?php
+    $cacheDriver = new \Doctrine\Common\Cache\ApcuCache();
+    $cacheDriver->save('cache_id', 'my_data');
+
```
Comment (greg0ire) [inline docs/en/reference/caching.rst]: I'd put it above the doc for APC, which is deprecated (maybe add a note about that, or even consider removing it?)

File: docs/en/reference/caching.rst [hunk 2]
```diff
@@ -63,6 +63,24 @@ by itself.
     $cacheDriver = new \Doctrine\Common\Cache\ApcCache();
     $cacheDriver->save('cache_id', 'my_data');
 
+APCu
+~~~
```
Comment (greg0ire) [inline docs/en/reference/caching.rst]: Please add one tilde here

**Comment by greg0ire** [comment]
APC is used in examples for the query cache or the result cache, should be replaced with APCu IMO: https://github.com/doctrine/doctrine2/blob/cd11723e6310b5973f50e7d9819c58916721d290/docs/en/reference/caching.rst

**Comment by greg0ire** [comment]
Same here: https://github.com/doctrine/doctrine2/blob/cd11723e6310b5973f50e7d9819c58916721d290/docs/en/reference/metadata-drivers.rst

File: docs/en/reference/caching.rst [hunk 3]
```diff
@@ -60,7 +60,25 @@ by itself.
 .. code-block:: php
 
     <?php
-    $cacheDriver = new \Doctrine\Common\Cache\ApcCache();
+    $cacheDriver = new \Doctrine\Common\Cache\ApcuCache();
```
Comment (greg0ire) [inline docs/en/reference/caching.rst]: You don't want to change that one.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Pull request merged**